exonum/src/node/consensus.rs

```diff
@@ -210,34 +206,40 @@ impl NodeHandler {
         }
 
         if self.state.block(&block_hash).is_none() {
-            let snapshot = self.blockchain.snapshot();
-            let schema = Schema::new(&snapshot);
+
             // Verify transactions
             let mut tx_hashes = Vec::new();
-            for raw in msg.transactions() {
-                if let Some(tx) = self.blockchain.tx_from_raw(raw) {
-                    let hash = tx.hash();
-                    if schema.transactions().contains(&hash) {
-                        error!(
-                            "Received block with already committed transaction, block={:?}",
-                            msg
-                        );
-                        return;
-                    }
-                    profiler_span!("tx.verify()", {
-                        if !tx.verify() {
-                            error!("Incorrect transaction in block detected, block={:?}", msg);
+            let mut fork = self.blockchain.fork();
+            {
+                let mut schema = Schema::new(&mut fork);
+
+                for raw in msg.transactions() {
+                    if let Some(tx) = self.blockchain.tx_from_raw(raw) {
+                        let hash = tx.hash();
+                        if schema.transactions().get(&hash).is_some() {
```

Comment:
Probably I'm missing something, but I'm not sure I understand how this check works now. If I understand correctly, transactions in the mempool are in the `transactions()` index, so it may be the case that some of them are uncommitted? Also, could it make sense to use `contains(&hash)` method instead of `get(&hash).is_some()`?